p2p/client.go

```diff
@@ -13,15 +13,22 @@ import (
 	"github.com/libp2p/go-libp2p-core/peer"
 	discovery "github.com/libp2p/go-libp2p-discovery"
 	dht "github.com/libp2p/go-libp2p-kad-dht"
-	"github.com/libp2p/go-libp2p/p2p/host/routed"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	routedhost "github.com/libp2p/go-libp2p/p2p/host/routed"
 	"github.com/multiformats/go-multiaddr"
+	"go.uber.org/multierr"
 
 	"github.com/lazyledger/optimint/config"
 	"github.com/lazyledger/optimint/log"
 )
 
-// reAdvertisePeriod defines a period after which P2P client re-attempt advertising namespace in DHT.
-const reAdvertisePeriod = 1 * time.Hour
+const (
+	// reAdvertisePeriod defines a period after which P2P client re-attempt advertising namespace in DHT.
+	reAdvertisePeriod = 1 * time.Hour
```

Comment:
That is so long because once they've advertized this it won't change?

Reply:
Exactly.